astropy/io/votable/tests/table_test.py

```diff
@@ -175,6 +179,80 @@ def test_write_with_format():
 def test_empty_table():
     votable = parse(
         get_pkg_data_filename('data/empty_table.xml'),
-        pedantic=False)
+        verify='warn')
     table = votable.get_first_table()
     astropy_table = table.to_table()  # noqa
+
+
+def test_verify_options(tmpdir):
+
+    # Start off by checking the default (ignore)
+
+    with catch_warnings(VOWarning) as w:
+        parse(get_pkg_data_filename('data/gemini.xml'))
+    assert len(w) == 0
+
+    # Then try the various explicit options
+
+    with catch_warnings(VOWarning) as w:
+        parse(get_pkg_data_filename('data/gemini.xml'), verify='ignore')
+    assert len(w) == 0
+
+    with catch_warnings(VOWarning) as w:
+        parse(get_pkg_data_filename('data/gemini.xml'), verify='warn')
+    assert len(w) == 25
+
+    with pytest.raises(VOWarning):
+        parse(get_pkg_data_filename('data/gemini.xml'), verify='exception')
+
+    # Make sure the pedantic option still works for now (pending deprecation)
+
+    with catch_warnings(VOWarning) as w:
+        parse(get_pkg_data_filename('data/gemini.xml'), pedantic=False)
+    assert len(w) == 25
+
+    with pytest.raises(VOWarning):
+        parse(get_pkg_data_filename('data/gemini.xml'), pedantic=True)
+
+    # Make sure that the default behavior can be set via configuration items
+
+    with conf.set_temp('verify', 'ignore'):
+        with catch_warnings(VOWarning) as w:
+            parse(get_pkg_data_filename('data/gemini.xml'))
+        assert len(w) == 0
+
+    with conf.set_temp('verify', 'warn'):
+        with catch_warnings(VOWarning) as w:
+            parse(get_pkg_data_filename('data/gemini.xml'))
+        assert len(w) == 25
+
+    with conf.set_temp('verify', 'exception'):
+        with pytest.raises(VOWarning):
+            parse(get_pkg_data_filename('data/gemini.xml'))
+
+    # And make sure the old configuration item will keep working
+
+    with set_temp_config(tmpdir.strpath):
+
+        with open(tmpdir.join('astropy').join('astropy.cfg').strpath, 'w') as f:
+            f.write('[io.votable]\npedantic = False')
+
+        reload_config('astropy.io.votable')
+
+        with catch_warnings(VOWarning) as w:
+            parse(get_pkg_data_filename('data/gemini.xml'))
+        assert len(w) == 25
+
+        with open(tmpdir.join('astropy').join('astropy.cfg').strpath, 'w') as f:
+            f.write('[io.votable]\npedantic = True')
+
+        reload_config('astropy.io.votable')
+
+        with pytest.raises(VOWarning):
+            parse(get_pkg_data_filename('data/gemini.xml'))
+
+    reload_config('astropy.io.votable')
```

Comment:
What does this one do, exactly?

Reply:
Once we exit the context manager, the config dir changes back to the original directory so we need to reload again to get the original config back, I think. I'll double check and can add a comment.

Reply:
But this is no longer needed once the tests are refactored into a class...